### PR TITLE
Input parameter tables

### PR DESCRIPTION
**Pull request by leeyap**
Hi @kdorheim,

Here is some code that reads in a csv with input parameters from an ini file (not sure where this will live in the repo so I have it in `/vignettes/articles` for now) and produces two tables. 

Let me know what you think and if there are any changes you'd like me to make.

### CODE REVIEW

**Comment by kdorheim** [comment]
Hi Leeya it is looking good! some changes that would be nice but aren't essential include

* add a parameter and variable inputs section headers 
* delete the `Properties with asterisks (*) are optional.` line because that information is included in the table 
* Table changes that would be nice but are not essential. Format the table so that the column titles are always available when users are viewing any part of the table? Could the section portion of the table be grouped? 
* @bpbond what do you think about switching our the column name from parameter to input variable in table number 2?

**Comment by leeyap** [comment]
Hi @kdorheim,

Sure thing, I can make these changes. When you say
>  Could the section portion of the table be grouped?

Do you mean to create a new table for each section, or something else?

Also, when I knit the file locally, I can see the column names even if I scroll down - is it not working on your end?

Thankss!

**Comment by kdorheim** [comment]
@leeyap the best way to preview what the documentation will appear like online is to preview it with the `pkgdown::build_articles()` function, whatever defaults are set up in the pkgdown template must be overwriting the preview column names feature. 

Yes I think separate tables for the parameters, biome specific parameters, and input time series would be helpful. 

As for grouping the table by section, I was thinking of something along the line of [like this](https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html#Grouped_Columns__Rows).

**Comment by leeyap** [comment]
@kdorheim hm, I can't seem to figure out how to override the `pkgdown` override - the internet says manipulating the yaml file, but I'm not sure if that applies here, and I saw another [post](https://github.com/r-lib/pkgdown/issues/1544) that a `kable` feature isn't supporting by `pkgdown` at all. 

For grouping, that sounds good - were you thinking that each section is its own group, with the remaining columns grouped under them? Having a hard time visualizing this, sorry 🙃

File: vignettes/articles/input_params.csv [hunk 1]
```diff
@@ -0,0 +1,223 @@
+section,parameter,biome-specific,time-variant,required,default,units,description
```
Comment (kdorheim) [inline vignettes/articles/input_params.csv:1]: Looking at the generated html tables the I see some unit entries`W yr m<96>2 C Tg<96>1` when I open the csv file there are `W yr mñ2 C Tgñ1` entries could you make sure that the entires that the units are formatted properly?

File: vignettes/articles/InputFiles.Rmd [hunk 2]
```diff
@@ -63,21 +75,15 @@ lucEmissions=csv:input/emissions/ssp245_emissions.csv
 
 In this case, the `input/emissions/ssp245_emissions.csv` file is parsed and the data in column `lucEmissions` are assigned to the model variable, with a required `Date` column providing the time index.
 
-**Table of variable inputs (with a time index)**
```
Comment (kdorheim) [inline vignettes/articles/InputFiles.Rmd:66]: Add a comment about how the "(csv)" entry in the Default column indicates that the values are passed in via a path to a csv file

File: vignettes/articles/InputFiles.Rmd [hunk 3]
```diff
@@ -18,35 +18,47 @@ run_name=ssp245
 Here, comments start with a semicolon (`;`); *section names* (`core`) are enclosed in square brackets; and a *property* (`run_name`) is assigned a *value* (`ssp245`). Properties are guaranteed to be unique within a particular section, and all sections match one-to-one with a model [component](ComponentAPI.html).
 
 
-**Table of model parameters**
```
Comment (kdorheim) [inline vignettes/articles/InputFiles.Rmd:21]: Please update [SSP245 file](/inst/input/hector_ssp245.ini) to [SSP245 file](https://github.com/JGCRI/hector/tree/master/inst/input/hector_ssp245.ini), note that this path will not work until the hector v3 branch is merged into master

**Review by kdorheim — CHANGES_REQUESTED** [review]
Some minor changes

**Comment by leeyap** [comment]
Also, where should the input csv file live?

**Comment by kdorheim** [comment]
@bpbond do you have thoughts on where the input param file should live? I guess if we are going to be using it in the testing or to generate some sort of table for the R interface it probably should not live in the vignettes/articles directory

**Comment by bpbond** [comment]
@kdorheim Apologies that I missed seeing this earlier.

Re where it should live: maybe in `data_raw/`? Then we can `usethis::use_data` to make it package data easily available to anyone?

**Comment by kdorheim** [comment]
Yes I think that the data-raw would be a good place for it to live

**Review by kdorheim — APPROVED** [review]

**Comment by kdorheim** [comment]
@leeyap please ping me to let me know when you are ready for feedback on your tests 👍

**Comment by leeyap** [comment]
@kdorheim first test is passing! The second test is not at the moment but let me take a stab at figuring out what's going on. Do you anticipate any more changes to the ini files or inputs that I should keep in mind?

**Review by kdorheim — APPROVED** [review]
Nicely done Leeya!

**Comment by leeyap** [comment]
@kdorheim thanks! Okay to merge?

**Comment by kdorheim** [comment]
@leeyap yes please do

**Pull request merged**